### PR TITLE
feat(backend): Managed Authentication

### LINKED ISSUE #2979: Managed User Authentication Backend
Give the Platform a Public/Private Key, Where they use Public Key to Sign users and redirect them to the frontend so it's exchanged to temporary JWT Token.

### PR DESCRIPTION
**Pull request by khaledmashaly**
## What does this PR do?

fixes #2979

### CODE REVIEW

File: packages/backend/src/app/database/postgres-connection.ts [hunk 1]
```diff
@@ -159,6 +160,7 @@ const getMigrations = (): (new () => MigrationInterface)[] => {
                 AddPlatform1697717995884,
                 AddCustomDomain1698077078271,
                 AddSigningKey1698602417745,
+                ManagedAuthnInitial1698700720482,
```
Comment (abuaboud) [inline packages/backend/src/app/database/postgres-connection.ts]: Let's move to all platforms specially the edits on project and user

File: packages/ee/shared/src/lib/managed-authn/managed-authn-requests.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Type, Static } from '@sinclair/typebox'
+
+export const ManagedAuthnRequestBody = Type.Object({
```
Comment (abuaboud) [inline packages/ee/shared/src/lib/managed-authn/managed-authn-requests.ts:3]: Why not ManagedAuth, or ManagedAuthentication?

**Review by abuaboud — CHANGES_REQUESTED** [review]

File: packages/ee/shared/src/lib/managed-authn/managed-authn-requests.ts (hunk 2 above)
Comment (khaledmashaly) [reply packages/ee/shared/src/lib/managed-authn/managed-authn-requests.ts:3]: authn is the defacto abbreviation for authentication, authz is for authorization

File: packages/backend/src/app/database/postgres-connection.ts (hunk 1 above)
Comment (khaledmashaly) [reply packages/backend/src/app/database/postgres-connection.ts]: done

**Pull request merged**